### ISSUE TITLE
Not all outputs need to be translated back to Julia

### ISSUE THREAD

**Issue by alex-lew** [probcomp/GenPyTorch.jl#1]
In some cases, an output of a neural net will not be used by Julia code, but rather fed directly as input to another neural net. For example, the hidden state of an RNN is passed straight through to the next cell of the RNN. In these cases, it is wasteful to copy the data to the CPU and convert to Julia Array type, only to later copy it back to the GPU and convert to tensor type again. 

We could have allow users to specify whether each output should be converted or not.

**Comment by JGameCreation** [comment]
I'm not sure if this is the reason but when feeding the outputs of a neural network into another TorchGenerativeFunction, effectively combining two of them, I experienced gradient computation to be 30 times slower than when I just combine those two networks directly in PyTorch and just have one TorchGenerativeFunction.
